plugins/modules/files/ini_file.py

```diff
@@ -311,6 +312,9 @@ def main():
     allow_no_value = module.params['allow_no_value']
     create = module.params['create']
 
+    if state == 'present' and not allow_no_value and not value:
+        module.fail_json("Parameter 'value' must not be empty if state=present and allow_no_value=False")
```

Comment:
This broke the possibility to add an option with empty value. Example:

```yaml
- name: Enable Mercurial evolve.
  ini_file:
    path:                     "{{ ansible_env.HOME }}/.hgrc"
    section:                  extensions
    option:                   evolve
    value:                    ""
```

Previously, it added a line `evolve =` and now it fails at this code line.

Setting the `allow_no_value` parameter is not correct because it would create the option without an equal sign.

empty value != no value

Reply:
@russoz can you take a look?

Reply:
yes, sure, will take a look at that

Reply:
Submitted #1972, continuing conversation there.